### DISCUSSION TITLE
useInfinateQuery not passing in { take, skip }

### DISCUSSION THREAD

**Discussion by lewisblackburn** [blitz-js/blitz#1448] [Q&A] [answered]
# useInfiniteQuery not passing in { take, skip }

Asking this at the risk of sounding dumb, I was looking at the docs on how to use the InfiniteQuery hook and copied the code straight over, however, I realised that it was pulling all the data and wasn't taking `take: n` into account. I could change the values of `take` and `skip` inside of the `getMovies.ts` file, but not pass them in.

> movies/pages/movies/index.tsx
```tsx
  const [movies, { isFetching, isFetchingMore, fetchMore, canFetchMore }] = useInfiniteQuery(
    getMovies,
    (page = { take: 1, skip: 0 }) => page,
    {
      getFetchMore: (lastGroup) => lastGroup.nextPage,
    }
  )

console.log(movies, movies[0].movies)
```

> movies/queries/getMovies.ts
```tsx
import db, { FindManyMovieArgs } from "db"

type GetMoviesInput = Pick<FindManyMovieArgs, "where" | "orderBy" | "skip" | "take">

export default async function getMovies(
  { where, orderBy, skip = 0, take }: GetMoviesInput,
  ctx: Ctx
) {
  ctx.session.authorize()

  const movies = await db.movie.findMany({
    where,
    orderBy,
    take,
    skip,
  })

  const count = await db.movie.count()
  const hasMore = typeof take === "number" ? skip + take < count : false
  const nextPage = hasMore ? { take, skip: skip + take! } : null

  return {
    movies,
    nextPage,
    hasMore,
    count,
  }
}
```

**Comment by flybayer** [comment, marked as answer]
What blitz version are you on?

There was a bug with this on v0.25.0-canary.1 that's fixed in v0.25.0-canary.2

**Reply by lewisblackburn** [reply]
Yeah I checked that and I upgraded to v0.25.0-canary.5. I'm not sure whether this is the same issue as that or not.

**Reply by lewisblackburn** [reply]
Nevermind, It was the same issue and I just needed to create a new blitz project on the same version!
